lib/tasks/manageiq/ui_tasks.rake

```diff
@@ -24,6 +24,100 @@ namespace :update do
+        #
+        # * Redistributions in binary form must reproduce the above copyright notice,
+        #   this list of conditions and the following disclaimer in the documentation
+        #   and/or other materials provided with the distribution.
+        #
+        # THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+        # AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+        # IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+        # DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+        # FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+        # DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+        # SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+        # CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+        # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+        # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+        disk_usage = begin
+                       disk_size_path = if path.symlink?
+                                          path.dirname.join(path.readlink)
+                                        else
+                                          path
+                                        end
+
+                       if disk_size_path.directory?
+                         scanned_files = Set.new
+                         disk_usage = 0
+                         path.find do |f|
+                           if f.directory?
+                             disk_usage += f.lstat.size
+                           else
+                             # use Pathname#lstat instead of Pathname#stat to get info of symlink itself.
+                             stat = f.lstat
+                             file_id = [stat.dev, stat.ino]
+                             # count hardlinks only once.
+                             unless scanned_files.include?(file_id)
+                               disk_usage += stat.size
+                               scanned_files.add(file_id)
+                             end
+                           end
+                         end
+                       else
+                         disk_usage = path.lstat.size
+                       end
+
+                       total_savings += disk_usage
+                       disk_usage.to_f / 1_048_576
+                     end
+
+        puts "Removing #{path}... (#{'%.1f' % disk_usage}MB)"
+        FileUtils.rm_rf path, :secure => true, :noop => !!ENV["DRY_RUN"]
+      end
+    end
+
+    total = if total_savings >= 1_073_741_824
+              "#{'%.1f' % (total_savings.to_f / 1_073_741_824)}GB"
+            else
+              "#{'%.1f' % (total_savings.to_f / 1_048_576)}MB"
+            end
+
+    puts "\nTotal Savings: #{total}"
```

Comment:
Do we care about this?  I wonder if it's worth having it considering that we have to pull over the license.

Reply:
> Do we care about this?

The savings portion of the code or the entire PR in general?

:trollface: (sort of)

To your actual question, to some degree I think it shines light on to a major issue (to me at least) of bloat that can happen in the git gems and possibly wanting to re-structure where we put these assets so they aren't ending up being dumped into the git gem dirs.

That said, does it actually help with the effectiveness of the script?  No.  Also, it probably makes it slower, but I thought it was neat and agree that it probably isn't worth dragging in a license for that reason alone.

* * *

To the other question, can we live without this?  Probably.  I had a non-trival amount of space sucked up by `node-modules` dirs (about 5 gigs) prior to making this script, and figured I could write something that people might find useful and determine if it was for more folks than myself.

As discussed with Chris, there are other ways that this can be accomplished, some of which don't even require this script (`bundle clean`, for example) and so not merging this for that reason alone seems pretty reasonable.  Also, we can do this in a way that doesn't require polluting the production build by having an ancillary gem added to peoples `bundle` via `bundler.d` that injects this task as well, so I would be cool with going that route as well.